tests/operator_test.go

```diff
@@ -864,6 +865,67 @@ var _ = Describe("ALL Operator tests", func() {
 					}, 2*time.Minute, 1*time.Second).Should(BeNumerically("==", expectedIncomplete))
 				}
 			})
+
+			It("DataImportCron failing metric expected value when patching DesiredDigest annotation with junk sha256 value", func() {
+				if !tests.IsPrometheusAvailable(f.ExtClient) {
+					Skip("This test depends on prometheus infra being available")
+				}
+				var url string
+				var err error
+				numCrons := 2
+				retentionPolicy := cdiv1.DataImportCronRetainAll
+				trustedRegistryURL := fmt.Sprintf(utils.TrustedRegistryURL, f.DockerPrefix)
+				originalMetricVal := getMetricValue("kubevirt_cdi_dataimportcron_not_up_to_date_total")
+				expectedFailingCrons := originalMetricVal + numCrons
+				if utils.IsOpenshift(f.K8sClient) {
+					url = externalRegistryURL
+				} else {
+					url = trustedRegistryURL
+					err = utils.AddInsecureRegistry(f.CrClient, url)
+					Expect(err).To(BeNil())
+
+					hasInsecReg, err := utils.HasInsecureRegistry(f.CrClient, url)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(hasInsecReg).To(BeTrue())
+					defer utils.RemoveInsecureRegistry(f.CrClient, url)
+				}
+
+				for i := 0; i < numCrons; i++ {
+					cron := NewDataImportCron(fmt.Sprintf("cron-test-%d", i), "5Gi", scheduleOnceAYear, fmt.Sprintf("datasource-test-%d", i), cdiv1.DataVolumeSourceRegistry{URL: &url, PullMethod: &registryPullNode}, retentionPolicy)
+					By(fmt.Sprintf("Create new DataImportCron %s", url))
+					cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Create(context.TODO(), cron, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					var currentImportDv string
+					By("Wait for CurrentImports DataVolumeName update")
+					Eventually(func() bool {
+						cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(f.Namespace.Name).Get(context.TODO(), cron.Name, metav1.GetOptions{})
+						Expect(err).ToNot(HaveOccurred())
+						if len(cron.Status.CurrentImports) == 0 {
+							return false
+						}
+						currentImportDv = cron.Status.CurrentImports[0].DataVolumeName
+						return currentImportDv != ""
```

Comment:
why not simply wait for condition UpToDate = true?

Reply:
Right, I guess that indirectly tells me DV is succeeded, thanks